File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    k = {"code" : 0, "date" : 1, "maximum" : 2, "remain" : 3}
    
    answer = []
    for d in data:
        if d[k[ext]] < val_ext:
            answer.append(d)
    return sorted(answer, key=lambda x : x[k[sort_by]])

File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.java
import java.util.*;

class Solution {
    
    public int[][] solution(int[][] data, String ext, int val_ext, String sort_by) {
        Map<String, Integer> map = new HashMap<>();
        map.put("code", 0); map.put("date", 1);
        map.put("maximum", 2); map.put("remain", 3);
        
        List<int[]> list = new ArrayList<>();
        for (int[] d : data) {
            if (d[map.get(ext)] < val_ext) {
                list.add(d);
            }
        }
        
        list.sort((a, b) -> Integer.compare(a[map.get(sort_by)], b[map.get(sort_by)]));
        return list.toArray(new int[0][]);
    }
}